### PR TITLE
fixed warnings

### PR DESCRIPTION
**Pull request by huckl3b3rry87**
I fixed all of the warnings, but there is a test error that needs to be fixed before this package is tagged
```
===============================================================================

ERROR: LoadError: LoadError: Unsupported feature Jac
Stacktrace:
 [1] initialize(::Rosenbrock, ::Array{Symbol,1}) at /home/febbo/.julia/v0.6/MathProgBase/test/nlp.jl:272
 [2] loadproblem!(::KNITRO.KnitroMathProgModel, ::Int64, ::Int64, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Symbol, ::Rosenbrock) at /home/febbo/.julia/v0.6/KNITRO/src/KnitroSolverInterface.jl:80
 [3] rosenbrocktest(::KNITRO.KnitroSolver) at /home/febbo/.julia/v0.6/MathProgBase/test/nlp.jl:308
 [4] include_from_node1(::String) at ./loading.jl:569
 [5] include(::String) at ./sysimg.jl:14
 [6] include_from_node1(::String) at ./loading.jl:569
 [7] include(::String) at ./sysimg.jl:14
 [8] process_options(::Base.JLOptions) at ./client.jl:305
 [9] _start() at ./client.jl:371
while loading /home/febbo/.julia/v0.6/KNITRO/test/mathprogtest.jl, in expression starting on line 3
while loading /home/febbo/.julia/v0.6/KNITRO/test/runtests.jl, in expression starting on line 2
=====================================================================================================================================[ ERROR: KNITRO ]=====================================================================================================================================

failed process: Process(`/opt/julia-903644385b/bin/julia -Cx86-64 -J/opt/julia-903644385b/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /home/febbo/.julia/v0.6/KNITRO/test/runtests.jl`, ProcessExited(1)) [1]

```
`

### CODE REVIEW

**Comment by mlubin** [comment]
The way to fix the error is for KNITRO.jl not to ask for ``:Jac`` if there are no constraints present.

File: examples/jump_hs035_options.jl [hunk 1]
```diff
@@ -13,8 +13,8 @@ for m in [Model(solver=KnitroSolver()),
                             + 2.0*x[1]*x[2] + 2.0*x[1]*x[3])
     @constraint(m, x[1] + x[2] + 2.0*x[3] <= 3)
     solve(m)
-    @test_approx_eq_eps getvalue(x[1]) 1.3333333 1e-5
-    @test_approx_eq_eps getvalue(x[2]) 0.7777777 1e-5
-    @test_approx_eq_eps getvalue(x[3]) 0.4444444 1e-5
-    @test_approx_eq_eps getobjectivevalue(m) 0.1111111 1e-5
+    @test getvalue(x[1]) ≈ 1.3333333 atol=1.0e-5
```
Comment (mlubin) [inline examples/jump_hs035_options.jl]: Have you tested on Julia 0.5? I don't believe this syntax is present there

**Comment by sylvainmouret** [comment]
It looks like KnitroSolverInterface.jl does not currently support bound-constrained NLP problems.

File: examples/jump_hs035_options.jl (hunk 1 above)
Comment (huckl3b3rry87) [inline examples/jump_hs035_options.jl]: @mlubin no, I did not. I guess that we either need to not support 0.5 and move to 0.6 or have a check in the code for the version number and run different code based off of that. What do you prefer?

**Comment by huckl3b3rry87** [comment]
I updated `loadproblem!()` using `Ipopt.jl` as a reference basically the top few lines now look like this:
```
# generic nonlinear interface
function loadproblem!(m::KnitroMathProgModel,
                      numVar::Int,
                      numConstr::Int,
                      x_l, x_u, g_lb, g_ub,
                      sense::Symbol,
                      d::AbstractNLPEvaluator)

    features = features_available(d)
    has_hessian = (:Hess in features)
    init_feat = [:Grad]
    has_hessian && push!(init_feat, :Hess)
    numConstr > 0 && push!(init_feat, :Jac)

    initialize(d, init_feat)
    Ihess, Jhess = has_hessian ? hesslag_structure(d) : (Int[], Int[])
    Ijac, Jjac = numConstr > 0 ? jac_structure(d) : (Int[], Int[])

    m.nnzJ = length(Ijac)
    m.nnzH = length(Ihess)
    jac_tmp = Array{Float64}(m.nnzJ)
    hess_tmp = Array{Float64}(m.nnzH)
```
But I am getting a segfault error

File: examples/jump_hs035_options.jl (hunk 1 above)
Comment (mlubin) [inline examples/jump_hs035_options.jl]: just use ``isapprox`` instead of `≈`

**Pull request merged**